R/gateUtil.R

```diff
@@ -109,13 +79,11 @@ commonGateCreate = function(body, name, gid,
   gateResp = basePost(path, body, list())
 
   if (createPopulation) {
-    populationResp = autoCreatePopulation(experimentId, name, gid, parentPopulationId)
-    return(list(
-      gate = gateResp,
-      population = populationResp
-    ))
+    gateResp = basePost(path, body, params=list("createPopulation"='true'))
   } else {
-    return(list(gate = gateResp))
+    gateResp = basePost(path, body, list())
+
+  return(list(gate = gateResp))
```

Comment:
nit: indentation on this line, and I think you're missing a `return` path from the `if (createPopulation)` block